codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/glacier/GlacierChecksum.kt

```diff
@@ -0,0 +1,41 @@
+package software.amazon.smithy.aws.swift.codegen.customization.glacier
+
+import software.amazon.smithy.aws.swift.codegen.middleware.AWSSigningMiddleware
+import software.amazon.smithy.aws.swift.codegen.middleware.Sha256TreeHashMiddleware
+import software.amazon.smithy.aws.swift.codegen.sdkId
+import software.amazon.smithy.aws.traits.auth.UnsignedPayloadTrait
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.SwiftSettings
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareStep
+import software.amazon.smithy.swift.codegen.middleware.OperationMiddleware
+import software.amazon.smithy.swift.codegen.model.expectShape
+import software.amazon.smithy.swift.codegen.model.hasTrait
+
+/**
+ * Adds a middleware for Glacier to autofill accountId when not set
+ * See: https://github.com/awslabs/aws-sdk-swift/issues/208
+ * See also: https://docs.aws.amazon.com/amazonglacier/latest/dev/checksum-calculations.html
+ */
+class GlacierChecksum : SwiftIntegration {
+    override fun enabledForService(model: Model, settings: SwiftSettings) =
+        model.expectShape<ServiceShape>(settings.service).sdkId.equals("Glacier", ignoreCase = true)
```

Comment:
We should avoid using `ignoreCase` -- can we just explicitly list out all the services this Integration applies to?

Reply:
I guess why should we avoid using it?

Reply:
Other than a relatively negligible performance hit, I think using ignoreCase seems less direct.  Reading it with `ignoreCase = true` makes me think there we are targeting two services: one named `GLACIER` and one named `Glacier`.

Reply:
addressed